src/init/starship.tcsh

```diff
@@ -0,0 +1,6 @@
+setenv STARSHIP_SHELL tcsh;
+setenv STARSHIP_SESSION_KEY `(::STARSHIP:: session)`;
+alias precmd 'set STARSHIP_CMD_STATUS = $status;set STARSHIP_PATH = ::STARSHIP::;jobs > $STARSHIP_TCSH_TMPFILE;set NUM_JOBS=`(cat $STARSHIP_TCSH_TMPFILE | wc -l)`;rm $STARSHIP_TCSH_TMPFILE;set STARSHIP_END_TIME = `$STARSHIP_PATH time`;set STARSHIP_DURATION = 0;if ( $STARSHIP_START_TIME != -1 ) @ STARSHIP_DURATION = $STARSHIP_END_TIME - $STARSHIP_START_TIME;set prompt = "`($STARSHIP_PATH prompt --status $STARSHIP_CMD_STATUS --jobs $NUM_JOBS --cmd-duration $STARSHIP_DURATION)`";set STARSHIP_START_TIME = -1;';
```

Comment:
I don't really like the jobs workaround here. I'd prefer if you removed it and added a warning in the docs about it.

Also: The other shell inits preserve the old `precmd` and `postcmd` hooks. Is this also possible in tcsh?

Reply:
1. I agree, it's a messy workaround.
2. I have added this functionality to the script.